src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js

```diff
@@ -207,6 +209,75 @@ const SubmissionWorkflowsTable = ({ workspace, submission }) => {
   ])
 }
 
+const SubmissionWorkflowIOTable = ({ type, inputOutputs }) => {
+  const [textFilter, setTextFilter] = useState('')
+  const [sort, setSort] = useState({ field: 'task', direction: 'asc' })
+
+  const filteredInputOutputs = _.flow(
+    _.toPairs,
+    _.map(([name, value]) => ({
+      task: ioTask(name),
+      variable: ioVariable(name),
+      value
+    })),
+    _.filter(({ task, variable }) => {
+      return _.some(_.includes(textFilter), [task, variable])
+    }),
+    sort.field === 'task' ?
+      _.orderBy(['task', 'variable'], [sort.direction, sort.direction]) :
+      _.orderBy(sort.field, sort.direction)
+  )(inputOutputs)
+
+  return h(Fragment, [
+    div({ style: { margin: '1rem 0', display: 'flex', alignItems: 'center' } }, [
+      h(DelayedSearchInput, {
+        style: { marginRight: '2rem', flexBasis: 300, borderColor: colors.dark(0.55) },
+        placeholder: `Search ${type}`,
+        'aria-label': `Search ${type}`,
+        onChange: setTextFilter,
+        value: textFilter
+      }),
+      h(Link, { onClick: () => downloadIO(inputOutputs, type) }, ['Download JSON'])
+    ]),
+    div({ style: { flex: `1 0 ${(1 + _.min([inputOutputs.length, 5.5])) * tableRowHeight}px` } }, [
+      h(AutoSizer, [({ width, height }) => h(FlexTable, {
+        'aria-label': `Submission ${type}`,
+        width, height, sort,
+        headerHeight: tableRowHeight,
+        rowHeight: tableRowHeight,
+        rowCount: filteredInputOutputs.length,
+        noContentMessage: `No matching ${type}`,
+        columns: [
+          {
+            size: { basis: 350, grow: 0 },
+            field: 'task',
+            headerRenderer: () => h(Sortable, { sort, field: 'task', onSort: setSort }, ['Task name']),
+            cellRenderer: ({ rowIndex }) => {
+              const { task } = filteredInputOutputs[rowIndex]
+              return h(TextCell, [task])
+            }
```

Comment:
This reads a bit easier for me. Your call.
```suggestion
            cellRenderer: ({ rowIndex }) => h(TextCell, [filteredInputOutputs[rowIndex].task])
```

Reply:
Done in [af85b77](https://github.com/DataBiosphere/terra-ui/pull/3214/commits/af85b777e297ee2a8c19bafadf772df78bbcdb6d).